docs/installation_guide.md

```diff
@@ -1,292 +1,316 @@
+Harbor can be installed by one of two approaches: 
+
+- **Online installer:** The installer downloads Harbor's images from Docker hub. For this reason, the installer is very small in size.
+
+- **Offline installer:** Use this installer when the host does not have an Internet connection. The installer contains pre-built images so its size is larger.
+
+
+All installers can be downloaded from the **[official release](https://github.com/vmware/harbor/releases)** page. 
+
+This guide describes the steps to install and configure Harbor by using the online or offline installer. The installation processes are almost the same. 
+
+If you run a previous version of Harbor, you may need to migrate the data to fit the new database schema. For more details, please refer to **[Data Migration Guide](migration_guide.md)**.
+
+In addition, the deployment instructions on Kubernetes has been created by the community. Refer to set up [Harbor on Kubernetes](kubernetes_deployment.md) for details.
+
+## Prerequisites for the target host
+Harbor is deployed as several Docker containers, and, therefore, can be deployed on any Linux distribution that supports Docker. The target host requires Python, Docker, and Docker Compose to be installed.  
+* Python should be version 2.7 or higher.  Note that you may have to install Python on Linux distributions (Gentoo, Arch) that do not come with a Python interpreter installed by default  
+* Docker engine should be version 1.10 or higher.  For installation instructions, please refer to: https://docs.docker.com/engine/installation/
+* Docker Compose needs to be version 1.6.0 or higher.  For installation instructions, please refer to: https://docs.docker.com/compose/install/
+
+## Installation Steps
+
+The installation steps boil down to the following
+
+1. Download the installer;
+2. Configure **harbor.cfg**;
+3. Run **install.sh** to install and start Harbor;
+
+
+#### Downloading the installer:
+
+The binary of the installer can be downloaded from the [release](https://github.com/vmware/harbor/releases) page. Choose either online or offline installer. Use *tar* command to extract the package.
+
+Online installer:
+```
+    $ tar xvf harbor-online-installer-<version>.tgz
+```
+Offline installer:
+```
+    $ tar xvf harbor-offline-installer-<version>.tgz
+```
+
+#### Configuring Harbor
+Configuration parameters are located in the file **harbor.cfg**. 
+
+There are two categories of parameters in harbor.cfg, **required parameters** and **optional parameters**.  
+* **required parameters**: These parameters are required to be set in the configuration file, and they will take effect if user updates them in harbor.cfg, rerun the ```install.sh``` script to reinstall Harbor.
+* **optional parameters**: These parameters are optional, and only take effect in the initial installation, user can leave them empty and update them on UI after Harbor is started, subsequential update to these parameters in ```harbor.cfg``` will be ignored.
```

Comment:
These parameters are optional, and only take effect in the initial installation.  The user can leave them blank and update them on Web UI after Harbor is started. Subsequent update to these parameters in ```harbor.cfg``` will be ignored.